### PR TITLE
Add minimal LaunchTemplate to support EBS volume configuration

### PR DESCRIPTION
**Pull request by bacek**
This is initial/minimal version which I need right now to adjust AWS Batch environment. I'll implement more features bit later.

### CODE REVIEW

**Comment by paul-thompson-helix** [comment]
Looks good - watch out to map fields against the terraform AWS provider v2.70.

I thought launch template was an alternative to launch-configuration?
Would have expected some kind of opt-in or switch at the library level to choose between them? (Correct me if I'm wrong).

**Comment by bacek** [comment]
Template is alternative for configuration. Opt-in is in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#launch_template. You can specify one or another.

**Comment by paul-thompson-helix** [comment]
In hx-terraform is it just the case that both launch template and launch configuration are optional, and one should ensure to fill-in one or the other?

**Comment by bacek** [comment]
Correct. Same approach as for `name` or `name_prefix` use.

File: tools/gen-providers.ts [hunk 1]
```diff
@@ -1140,6 +1140,68 @@ const launch_configuration: RecordDecl = {
...
+    // placement - The placement of the instance. See Placement below for more details.
+    // ram_disk_id - The ID of the RAM disk.
+    // security_group_names - A list of security group names to associate with. If you are creating Instances in a VPC, use vpc_security_group_ids instead.
+    // vpc_security_group_ids - A list of security group IDs to associate with. Conflicts with network_interfaces.security_groups
+    // tag_specifications - The tags to apply to the resources during launch. See Tag Specifications below for more details.
+    // tags - (Optional) A map of tags to assign to the launch template. If configured with a provider default_tags configuration block present, tags with matching keys will overwrite those defined at the provider-level.
+    // user_data - The Base64-encoded user data to provide when launching the instance.
+    // hibernation_options - The hibernation options for the instance. See Hibernation Options below for more details.
+    // enclave_options - (Optional) Enable Nitro Enclaves on launched instances. See Enclave Options below for more details.
```
Comment (paul-thompson-helix) [inline tools/gen-providers.ts:1201]: As of current hx-terrform you would want to refer only to the (old) version of AWS provider.

https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/resources/launch_template#argument-reference

(doesn't have enclave_options)

File: tools/gen-providers.ts [hunk 2]
```diff
@@ -1140,6 +1140,68 @@ const launch_configuration: RecordDecl = {
...
+const launch_template_block_device_mapping_ebs: RecordDecl = {
+  name: 'launch_template_block_device_mapping_ebs',
+  fields: [
+    optionalField('delete_on_termination', BOOLEAN),
+    // optionalField('encrypted', BOOLEAN),
+    optionalField('iops', NUMBER),
+    // kms_key_id
+    optionalField('snapshot_id', STRING),
+    optionalField('throughput', NUMBER),
+    optionalField('volume_type', enumType(["standard", "gp2", "io1", "io2", "sc1", "st1"])),
```
Comment (paul-thompson-helix) [inline tools/gen-providers.ts:1153]: As of current hx-terrform you would want to refer only to the (old) version of AWS provider.

https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/resources/launch_template#volume_type

(has fewer volume_type options)

File: tools/gen-providers.ts (hunk 1 above)
Comment (bacek) [reply tools/gen-providers.ts:1201]: it's commented out for future use.

File: tools/gen-providers.ts (hunk 2 above)
Comment (bacek) [reply tools/gen-providers.ts:1153]: I literally copied it from https://github.com/helix-collective/hx-terraform/blob/25e0e27425455d295f4b1d6f468c590e62a892a5/tools/gen-providers.ts#L42

**Review by paul-thompson-helix — APPROVED** [review]
LGTM% watch out for use of values from later AWS provider.

**Pull request merged**